src/ui/public/timepicker/timepicker.js

```diff
@@ -76,23 +76,23 @@ module.directive('kbnTimepicker', function (refreshIntervals) {
       };
 
       $scope.absolute = {
-        from: moment(),
-        to: moment()
+        from: moment().toISOString(),
+        to: moment().toISOString()
       };
 
       $scope.units = timeUnits;
 
       $scope.relativeOptions = relativeOptions;
 
       $scope.$watch('from', function (date) {
-        if (moment.isMoment(date) && $scope.mode === TIME_MODES.ABSOLUTE) {
-          $scope.absolute.from = date;
+        if (date && $scope.mode === TIME_MODES.ABSOLUTE) {
+          $scope.absolute.from = moment.isMoment(date) ? date.toISOString() : date;
         }
       });
 
       $scope.$watch('to', function (date) {
-        if (moment.isMoment(date) && $scope.mode === TIME_MODES.ABSOLUTE) {
-          $scope.absolute.to = date;
+        if (date && $scope.mode === TIME_MODES.ABSOLUTE) {
+          $scope.absolute.to = moment.isMoment(date) ? date.toISOString() : date;
```

Comment:
If `date` is not a moment object, we simply return it. Could `date` be a native javascript date? Seems we should ensure we're returning a string no matter what.